Include list of merged pull requests in release notes

corneliusweig opened the pull request:
As mentioned in https://github.com/kubernetes-sigs/krew/issues/324#issuecomment-526905406

ferhatelmas commented:
why isn't it extracted to its own repo and maintained there?

It's simple but very likely to diverge this way.

ahmetb commented:
Extra repo mgmt is currently more costly, especially for standalone tools like this.

ahmetb commented on hack/make-release-notes.sh:
```diff
@@ -38,6 +42,10 @@ echo "Installation"
 echo "------------"
 echo "To install this release, refer to the instructions at ${readme}."
 echo
+echo "Merged pull requests"
```
I think this is definitely lower priority than contributors list + release asset links.

Maybe move this to the end?

And maybe if possible we should make it collapsible, like:

<details>
<summary>Pull Requests</summary>
<pre>
Html list here
</pre>
</details>

with

```
<details>
<summary>...</summary>
<pre>
...
</pre>
</details>
```

corneliusweig replied:
Yes, PRs have low priority, so I moved it to the end. I think it even works with markdown in the collapsed region, if a blank line is inserted. I haven't checked how the generated output renders on github though...

ferhatelmas commented:
> Extra repo mgmt is currently more costly, especially for standalone tools like this.

@ahmetb thanks, that's fair but how new changes in tools are propagated between repos right now?

ahmetb commented:
Looks like this tool is coming from minikube repo. In that case, there potentially could be a common repo, but again, making the case for it could be harder than copying + freezing the code.

In this case, as long as the code works and delivers the same result, we don't need to get new updates to it necessarily. We might even deem the tool unnecessary before it starts breaking.

corneliusweig commented:
@ferhatelmas I _think_ this comes from minikube, however I'm not even 100% sure here, because the same code is found in several repos.
As to upstream changes: we should be safe here, because github has a stable API, so the script will not suddenly stop working. Besides, we don't want another output formatting, so the script simply does what it does.

ahmetb commented on hack/release_notes/go.mod:
```diff
@@ -0,0 +1,10 @@
+module sigs.k8s.io/krew/hack/release_notes
```
Idea: why not make this a sub-module on minikube that we can `go get`?

corneliusweig replied:
Now that's a good idea! I'll open an issue and see if they are open for this.

ahmetb commented on hack/tools/go.mod:
```diff
@@ -0,0 +1,5 @@
+
+go 1.12
+
+require github.com/corneliusweig/release-notes v0.0.0-20191014214505-0be5c7c66752
```
maybe tag a release?

ahmetb replied:
let's say I have this

v0.3.1-alpha.1 -->  v0.3.1 --> v0.3.2 --> v0.3.3-alpha.1 --> v0.3.3

when I generate notes for v0.3.3, does it compute since v0.3.2 –or since v0.3.3-alpha.1?

corneliusweig replied:
That would only include in the range v0.3.3-alpha.1 to present. But it totally makes sense to change that :)

ahmetb replied:
Yeah we’d only want changes between revisions went to the user.

corneliusweig replied:
I've introduced an option for the tool, so that it now calculates the difference since v0.3.2 and skips the alpha version. Also, I tagged a release :)

ahmetb commented on hack/tools/tools.go:
```diff
@@ -0,0 +1,17 @@
+
+package main
+
+import _ "github.com/corneliusweig/release-notes"
```
this is not great. :( introduces a go package here.

let's just do something like this in the script, we already do it elsewhere:

```
temp="$(mkdir -p)"
trap -- 'rm -rf "${temp}"'
cd "$temp"
go mod init foo
go get github.com/corneliusweig/release-notes
# mv to '$(go env GOPATH)/bin' or something like that
```

corneliusweig replied:
That's also fine with me.

corneliusweig replied:
should be fixed now.

corneliusweig commented:
@ahmetb I finally addressed your comments:

- alpha releases are now ignored by the external tool when determining the latest previous release
- the installation method no longer needs a sub-module.

ahmetb commented on hack/make-release-notes.sh:
```diff
@@ -34,6 +38,22 @@ download_assets=(
+
+  cd "$relnotes_dir"
+  go mod init foo
+  GOBIN="${SCRIPTDIR}" go install github.com/corneliusweig/release-notes
```
I recall the whole point of this mktemp trick was to pin it to a particular version :)

ahmetb commented:
I think this is good, two concerns:

- we don't pin to a version of release-notes tool
- we don't use the same semver library as that tool, basically we just need to be able to tell if https://godoc.org/k8s.io/apimachinery/pkg/util/version#Version.PreRelease string empty or not to be able to tell a stable version or not.

These can be addressed later if you want.

/lgtm
/approve
